mvp_example/src/main/java/com/distillery/android/blueprints/mvp/MainApplication.kt

```diff
@@ -0,0 +1,30 @@
+package com.distillery.android.blueprints.mvp
+
+import android.app.Application
+import com.distillery.android.blueprints.mvp.feature.todo.ToDoContractView
+import com.distillery.android.blueprints.mvp.feature.todo.ToDoPresenter
+import com.distillery.android.domain.FakeToDoRepository
+import com.distillery.android.domain.ToDoRepository
+import kotlinx.coroutines.CoroutineScope
+import org.koin.android.ext.koin.androidContext
+import org.koin.android.ext.koin.androidLogger
+import org.koin.core.context.startKoin
+import org.koin.core.module.Module
+import org.koin.dsl.module
+
+class MainApplication : Application() {
+    private val modules: Module = module {
+        single<ToDoRepository> { (scope: CoroutineScope) -> FakeToDoRepository(scope) }
+        single { (view: ToDoContractView) -> ToDoPresenter(view) }
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+
+        startKoin {
+            androidLogger()
+            androidContext(this@MainApplication)
+            modules(modules)
```

Comment:
Minor: I think we can even inline the module, it won't make readability worse for such a small number of providers:
```suggestion
            modules(module {
                single<ToDoRepository> { (scope: CoroutineScope) -> FakeToDoRepository(scope) }
                single { (view: ToDoContractView) -> ToDoPresenter(view) }
            })
```